### PR TITLE
Qt static build - installs missing static libraries and adds missing dependancies in pkgconfig and pri files

### PR DESCRIPTION
**Pull request by matteoceraico**
Installs:
- libbmalloc
- libWTF
- libJavaScriptCore
- libWebCore

Adds missing dependancies in:
- Qt5WebKit.pc
- qt_lib_webkit.pri

### CODE REVIEW

File: Source/WebKit/PlatformQt.cmake [hunk 1]
```diff
@@ -351,10 +351,32 @@ install(
...
+        set(PKGCONFIG_EXTRA_LIBS "${PKGCONFIG_EXTRA_LIBS} ${LIB_NAME}")
+        set(PRI_EXTRA_LIBS "${PRI_EXTRA_LIBS} -l${LIB_NAME}")
+    endforeach()
+endif ()
+
 ecm_generate_pkgconfig_file(
     BASE_NAME Qt5WebKit
     LIB_INSTALL_DIR "lib"
-    DEPS "Qt5Core Qt5Gui Qt5Network"
+    DEPS "Qt5Core Qt5Gui Qt5Network Qt5Sql${PKGCONFIG_EXTRA_LIBS}"
```
Comment (annulen) [inline Source/WebKit/PlatformQt.cmake]: We should not link users of standard build to Qt5Sql, it's a private dependency

File: Source/WebKit/PlatformQt.cmake [hunk 2]
```diff
@@ -364,11 +386,12 @@ ecm_generate_pri_file(
     LIB_NAME QtWebKit
     INCLUDE_INSTALL_DIR "include"
     INCLUDE_INSTALL_DIR2 "include/QtWebKit"
-    DEPS "core gui network"
-    RUNTIME_DEPS "sensors positioning qml quick webchannel sql core_private gui_private"
+    DEPS "core gui network sql"
```
Comment (annulen) [inline Source/WebKit/PlatformQt.cmake]: for shared build it's runtime dependency

File: Source/JavaScriptCore/CMakeLists.txt [hunk 3]
```diff
@@ -1312,10 +1312,17 @@ add_subdirectory(shell)
 WEBKIT_WRAP_SOURCELIST(${JavaScriptCore_SOURCES})
 WEBKIT_FRAMEWORK(JavaScriptCore)
 
-if (${JavaScriptCore_LIBRARY_TYPE} STREQUAL "SHARED")
+if (${JavaScriptCore_LIBRARY_TYPE} STREQUAL "SHARED" OR QT_STATIC_BUILD)
```
Comment (Vitallium) [inline Source/JavaScriptCore/CMakeLists.txt]: On Windows we always build JavaScriptCore as a static library. If this necessary for other systems (Linux and macOS), then it should be wrapped.

File: Source/WebKit/PlatformQt.cmake [hunk 4]
```diff
@@ -351,10 +351,32 @@ install(
         ${CMAKE_INSTALL_PREFIX}/include/QtWebKit/${PROJECT_VERSION}/QtWebKit/private
 )
 
+if (QT_STATIC_BUILD)
+    set(EXTRA_LIBS_NAMES "WTF" "JavaScriptCore" "WebCore" "xml2")
```
Comment (Vitallium) [inline Source/WebKit/PlatformQt.cmake]: `xml2` is not a valid library name on Windows. On Windows we have `libxml2` to match Apple and Cairo ports.

File: Source/WebKit/PlatformQt.cmake [hunk 5]
```diff
@@ -351,10 +351,32 @@ install(
...
+    if (NOT USE_SYSTEM_MALLOC)
+        list(APPEND EXTRA_LIBS_NAMES "bmalloc")
+    endif ()
+    if (ENABLE_XSLT)
+        list(APPEND EXTRA_LIBS_NAMES "xslt")
+    endif ()
+    if (APPLE)
+        list(APPEND EXTRA_LIBS_NAMES "icucore")
+    else ()
+        list(APPEND EXTRA_LIBS_NAMES "icu")
```
Comment (Vitallium) [inline Source/WebKit/PlatformQt.cmake]: Same here.

File: Source/bmalloc/CMakeLists.txt [hunk 6]
```diff
@@ -31,3 +31,10 @@ include_directories(${bmalloc_INCLUDE_DIRECTORIES})
 add_library(bmalloc STATIC ${bmalloc_SOURCES})
 target_link_libraries(bmalloc ${bmalloc_LIBRARIES})
 set_target_properties(bmalloc PROPERTIES COMPILE_DEFINITIONS "BUILDING_bmalloc")
+
+if (QT_STATIC_BUILD)
+    install(TARGETS bmalloc EXPORT WebKitTargets
```
Comment (Vitallium) [inline Source/bmalloc/CMakeLists.txt]: bmalloc is not ported to Windows yet.

File: Source/WebKit/PlatformQt.cmake [hunk 7]
```diff
@@ -381,9 +381,45 @@ install(
...
+set(WEBKITWIDGETS_PRI_RUNTIME_DEPS "sensors positioning widgets_private opengl sql core_private gui_private")
+if (QT_STATIC_BUILD)
+    if (MSVC)
+        set(LIB_PREFIX "lib")
+    endif ()
+    set(WEBKIT_PKGCONFIG_DEPS "${WEBKIT_PKGCONFIG_DEPS} Qt5Sql")
+    set(WEBKIT_PRI_DEPS "${WEBKIT_PRI_DEPS} sql")
+    set(WEBKITWIDGETS_PKGCONFIG_DEPS "${WEBKITWIDGETS_PKGCONFIG_DEPS} Qt5PrintSupport")
+    set(WEBKITWIDGETS_PRI_DEPS "${WEBKITWIDGETS_PRI_DEPS} printsupport")
+    set(EXTRA_LIBS_NAMES "WTF" "JavaScriptCore" "WebCore" "xml2")
```
Comment (annulen) [inline Source/WebKit/PlatformQt.cmake]: Order is wrong, should be

    set(EXTRA_LIBS_NAMES WebCore JavaScriptCore WTF xml2)

**Comment by annulen** [comment]
Superceded by #323

**Pull request closed without merging**